concent_api/core/payments/storage.py

```diff
@@ -20,10 +21,37 @@ class DatabaseTransactionsStorage(TransactionsStorage):
     database using Django models.
     """
 
-    def __init__(self, nonce: int, *args: List, **kwargs: Dict) -> None:
-        super().__init__(*args, **kwargs)
+    @transaction.atomic(using='control')
+    def init(self, network_nonce: int) -> None:
+        if not self._is_storage_initialized():
+            self._init_with_nonce(network_nonce)
+            return
+
+        #  If nonce stored in GlobalTransactionState is lower than network_nonce
+        #  this statement will update nonce in DatabaseTransactionStorage
+        if self._get_nonce() < network_nonce:
+            global_transaction_state = GlobalTransactionState.objects.select_for_update().get(
+                pk=0,
+            )
+            global_transaction_state.nonce = network_nonce
+            global_transaction_state.full_clean()
+            global_transaction_state.save()
+            return
 
-        assert isinstance(nonce, int)
+    @transaction.atomic(using='control')
+    def _is_storage_initialized(self) -> bool:
+        """
+        Should return False if this is the first time we try to use this
+        storage.
+        """
+        return bool(GlobalTransactionState.objects.filter(pk=0).exists())
```

Comment:
There is no need to cast it to `bool`, it will be `bool` already.

Reply:
Ok, thanks :) Done